File: StudetDataBackup/5. Statistical Inference/t_test_paired.R
setwd("C:/Training/Academy/Statistics (Python)/Datasets")
anorexia <- read.csv("anorexia.csv")

cont <- subset(anorexia, Treat=="Cont")
t.test(cont$Prewt, cont$Postwt, mu = 0, paired = T, alternative = "l")

cbt <- subset(anorexia, Treat=="CBT")
t.test(cbt$Prewt, cbt$Postwt, mu = 0, paired = T, alternative = "l")

ft <- subset(anorexia, Treat=="FT")
t.test(ft$Prewt, ft$Postwt, mu = 0, paired = T, alternative = "l")

############## Rheumatic #####################
rhum <- read.csv("Rheumatic.csv")
t.test(rhum$Before, rhum$After, paired = T, alternative = "l")
#or
t.test(rhum$After, rhum$Before, paired = T, alternative = "g")

################ Plaque ##################
library(tidyr)
plaque <- read.csv("Plaque.csv")

plaq_pivot <- pivot_wider(plaque, id_cols = c("Id","product"),
            names_from = "trt", values_from = "score")

P1 <- subset(plaq_pivot, product=="P1")
t.test(P1$Before, P1$After, paired = T, alternative = 'g')

P2 <- subset(plaq_pivot, product=="P2")
t.test(P2$Before, P2$After, paired = T, alternative = 'g')

P3 <- subset(plaq_pivot, product=="P3")
t.test(P3$Before, P3$After, paired = T, alternative = 'g')

File: StudetDataBackup/5. Statistical Inference/t_test_paired.py
import pandas as pd
from scipy.stats import ttest_rel

anorexia = pd.read_csv("anorexia.csv")

cont = anorexia[anorexia['Treat']=="Cont"]
ttest_rel(cont['Prewt'], cont['Postwt'], 
          alternative="less")

cbt = anorexia[anorexia['Treat']=="CBT"]
ttest_rel(cbt['Prewt'], cbt['Postwt'], 
          alternative="less")

ft = anorexia[anorexia['Treat']=="FT"]
ttest_rel(ft['Prewt'], ft['Postwt'], 
          alternative="less")

############## Rheumatic #####################
rhum = pd.read_csv("Rheumatic.csv")
ttest_rel(rhum['Before'], rhum['After'], 
          alternative="less")
# or
ttest_rel(rhum['After'], rhum['Before'], 
          alternative="greater")


################ Plaque ##################
plaque = pd.read_csv("Plaque.csv")

plaq_pivot = pd.pivot_table(plaque, 
                            index=['Id','product'],
                            columns='trt',
                            values='score')

plaq_pivot = plaq_pivot.reset_index()

P1 = plaq_pivot[plaq_pivot['product']=="P1"]
ttest_rel(P1['Before'], P1['After'], 
          alternative="greater")

P2 = plaq_pivot[plaq_pivot['product']=="P2"]
ttest_rel(P2['Before'], P2['After'], 
          alternative="greater")

P3 = plaq_pivot[plaq_pivot['product']=="P3"]
ttest_rel(P3['Before'], P3['After'], 
          alternative="greater")





